statsmodels/tsa/base/tsa_model.py

```diff
@@ -204,12 +210,13 @@ def _init_dates(self, dates=None, freq=None):
         has_index = index is not None
         date_index = isinstance(index, (DatetimeIndex, PeriodIndex))
         int_index = isinstance(index, Int64Index)
+        range_index = isinstance(index, RangeIndex)
         has_freq = index.freq is not None if date_index else None
         increment = Int64Index(np.arange(self.endog.shape[0]))
```

Comment:
In cases where RangeIndex is available, using that for `increment` prevents a memory allocation.

Reply:
This is a good idea. I'll try to play around with it to see if I can support pre-0.18 w/ `Int64Index` and post-0.18 with `RangeIndex` without too much special casing code.

Reply:
Another option is just `Index(range(self.endog.shape[0])``.  That defaults to Int64Index in older versions and RangeIndex in newer versions.

Reply:
Thanks!

Reply:
> Another option is just `Index(range(self.endog.shape[0])``. That defaults to Int64Index in older versions and RangeIndex in newer versions.

This doesn't seem to be true for me?

```python
import pandas as pd
print(pd.__version__)
print(pd.Index(range(3)))
```

```
0.22.0
Int64Index([0, 1, 2], dtype='int64')
```

Reply:
Maybe there is a difference between Python 2 and Python 3 behavior here.